starfish/codebook/codebook.py

```diff
@@ -126,6 +133,15 @@ def _empty_codebook(cls, code_names: Sequence[str], n_ch: int, n_round: int):
             )
         )
 
+    @classmethod
+    def _verify_version(cls, semantic_version_str: str) -> None:
+        version = Version(semantic_version_str)
+        if not (MIN_SUPPORTED_VERSION <= version <= MAX_SUPPORTED_VERSION):
+            raise ValueError(
+                f"version {version} not supported.  This version of the starfish library only "
+                f"supports codebook formats from {MIN_SUPPORTED_VERSION} to "
+                f"{MAX_SUPPORTED_VERSION}")
+
     @classmethod
     def from_code_array(
```

Comment:
I intended this constructor to slurp in an already-in-memory codebook. 

I feel like it could be cleaner to update it to demand a Dict that includes a correct version parameter and do the verification, as otherwise this method is unaware of any changes to the codeword fields of the codebook.

Reply:
I disagree.  Managing the format should be the responsibility of this class.  If you want something that deals with a parsed document, it should 1. be internal to this class (i.e., something prefixed with _), and 2. not be named `from_code_array` (maybe `_from_codebook_document`).

Reply:
Ok, please update the example to be consistent with the new spec (I think just "codebook -> mappings"?)

Reply:
> Ok, please update the example to be consistent with the new spec (I think just "codebook -> mappings"?)

um where?

Reply:
I landed.  I will fix whatever comments you're referring to in a subsequent PR.